src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/kafka/RecordHandler.java

```diff
@@ -575,17 +501,105 @@ private void doDumpPortDescriptionRequest(CommandMessage message) {
+                        .collect(Collectors.toList());
+
+                MeterDumpResponse response = MeterDumpResponse.builder()
+                        .switchId(switchId)
+                        .meterSpeakerData(meters)
+                        .build();
+                sender.accept(response);
+            }
+
+        } catch (UnsupportedSwitchOperationException e) {
+            logger.info("Meters not supported: {}", switchId);
+            sender.accept(new SwitchMeterUnsupported(switchId));
+        } catch (SwitchNotFoundException e) {
+            logger.info("Dumping switch meters is unsuccessful. Switch {} not found", switchId);
+            ErrorData errorData = anError(ErrorType.NOT_FOUND)
+                    .withMessage(e.getMessage())
+                    .withDescription(switchId.toString())
+                    .buildData();
+            sender.accept(errorData);
+        } catch (SwitchOperationException e) {
+            logger.error("Unable to dump meters", e);
+            ErrorData errorData = anError(ErrorType.NOT_FOUND)
+                    .withMessage(e.getMessage())
+                    .withDescription("Unable to dump meters")
+                    .buildData();
+            sender.accept(errorData);
+        }
+    }
+
+    private void doDumpGroupsForSwitchManagerRequest(CommandMessage message) {
+        SwitchId switchId = ((DumpGroupsForSwitchManagerRequest) message.getData()).getSwitchId();
+        dumpGroupsRequest(switchId, buildSenderToSwitchManager(message));
+    }
+
+    private void doDumpGroupsForFlowHsRequest(CommandMessage message) {
+        SwitchId switchId = ((DumpGroupsForFlowHsRequest) message.getData()).getSwitchId();
+        dumpGroupsRequest(switchId, buildSenderToFlowHs(message));
+    }
+
+    private void dumpGroupsRequest(SwitchId switchId, java.util.function.Consumer<MessageData> sender) {
+        try {
+            logger.debug("Loading installed groups for switch {}", switchId);
+
+            List<OFGroupDescStatsEntry> ofGroupDescStatsEntries = context.getSwitchManager()
+                    .dumpGroups(DatapathId.of(switchId.toLong()));
+
+            List<GroupSpeakerData> groups = ofGroupDescStatsEntries.stream()
+                    .map(group -> OfGroupConverter.INSTANCE.convertToGroupSpeakerData(group, switchId))
+                    .collect(Collectors.toList());
+
+            GroupDumpResponse response = GroupDumpResponse.builder()
+                    .groupSpeakerData(groups)
+                    .build();
+            sender.accept(response);
+        } catch (SwitchOperationException e) {
+            logger.error("Dumping of groups on switch '{}' was unsuccessful: {}", switchId, e.getMessage());
+            ErrorData errorData = anError(ErrorType.NOT_FOUND)
+                    .withMessage(e.getMessage())
+                    .withDescription("The switch was not found when requesting a groups dump.")
```

Comment:
Fair point, but I just left the exception handling block as it was, 
But I did look at the code and found out the the only custom exception that could occur is SwitchNotFoundException, maybe I should change the old logic in order to narrow the exception type and left only SwitchNotFoundException exception instead of more common SwitchOperationException.

I agree to add the switch Id to the exception description.